src/utils/commondir.ts

```diff
@@ -0,0 +1,16 @@
+// ported from https://github.com/substack/node-commondir
+export default function commondir (files: string[]) {
+	const res = files.slice(1).reduce((ps, file) => {
+		const xs = file.split(/\/+|\\+/);
+		let i;
+		for (
+			i = 0;
+			ps[i] === xs[i] && i < Math.min(ps.length, xs.length);
+			i++
+		);
+		return ps.slice(0, i);
+	}, files[0].split(/\/+|\\+/));
+
+	// Windows correctly handles paths with forward-slashes
+	return res.length > 1 ? res.join('/') : '/';
+};
```

Comment:
May be nice to make the variables a little more descriptive now that we have inlined this. `xs` e.g. seems to be something like a `path` or `pathSegements`, `ps` is something like a `commonPath`, and `res` could be a `resultingPath`. Just some ideas to get rid of these completely arbitrary names.

Reply:
updated names